src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -66,29 +59,18 @@ export function useTextureData(): TextureDataState {
         loading: false,
         textureData: await proxy.computeTextureData(
           transfer(typedValues, [typedValues.buffer]),
-          customDomain || dataDomain,
+          domain,
           hasLogScale,
           colorMap
         ),
       });
     })();
-  }, [
-    colorMap,
-    customDomain,
-    dataDomain,
-    hasLogScale,
-    proxy,
-    mergeState,
-    values,
-  ]);
+  }, [colorMap, domain, hasLogScale, proxy, mergeState, values]);
 
-  const { rows, cols } = useDims();
+  // Reset texture when dimensions change to avoid rendering glitch while computing new texture
   const dims = `${rows}x${cols}`;
-  const [prevDims, setPrevDims] = useState<string>(dims);
-
-  if (prevDims !== dims) {
-    setPrevDims(dims);
-    mergeState({ textureData: undefined });
+  if (dims !== state.prevDims) {
+    mergeState({ textureData: undefined, prevDims: dims });
   }
```

Comment:
I had meant to do this but forgot: the previous dimensions are now stored in the same state as the texture. This avoids an extra `useState`. I've also added a comment.